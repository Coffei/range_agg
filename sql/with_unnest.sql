-- It combines with UNNEST
-- to implement the temporal database "coalesce" function
-- (see Snodgrass 6.5.2):
SELECT  room_id, t2.booked_during
FROM    (
          SELECT  room_id, range_agg(booked_during, true, true) AS booked_during
          FROM    reservations
          GROUP BY room_id
        ) AS t1,
        UNNEST(t1.booked_during) AS t2(booked_during)
ORDER BY room_id, booked_during
;
