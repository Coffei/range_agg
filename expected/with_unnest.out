-- It combines with UNNEST
-- to implement the temporal database "coalesce" function
-- (see Snodgrass 6.5.2):
SELECT  room_id, t2.booked_during
FROM    (
          SELECT  room_id, range_agg(booked_during, true, true) AS booked_during
          FROM    reservations
          GROUP BY room_id
        ) AS t1,
        UNNEST(t1.booked_during) AS t2(booked_during)
ORDER BY room_id, booked_during
;
 room_id |      booked_during      
---------+-------------------------
       1 | [07-01-2018,07-14-2018)
       1 | [07-20-2018,07-22-2018)
       2 | [07-01-2018,07-03-2018)
       5 | [07-01-2018,07-03-2018)
       6 | [07-01-2018,07-10-2018)
       7 | [07-01-2018,07-14-2018)
(6 rows)

